src/data/bucket/symbol_bucket.js

```diff
@@ -61,17 +60,18 @@ const symbolInterfaces = {
         ]
     },
     collisionBox: { // used to render collision boxes for debugging purposes
-        layoutVertexArrayType: createVertexArrayType([
+        layoutAttributes: [
             {name: 'a_pos',     components: 2, type: 'Int16'},
             {name: 'a_extrude', components: 2, type: 'Int16'},
             {name: 'a_data',    components: 2, type: 'Uint8'}
-        ]),
+        ],
         elementArrayType: createElementArrayType(2)
     }
 };
 
-function addVertex(array, x, y, ox, oy, tx, ty, minzoom, maxzoom, labelminzoom, labelangle) {
-    array.emplaceBack(
+function addVertex(array, x, y, ox, oy, tx, ty, sizeData, minzoom, maxzoom, labelminzoom, labelangle) {
+    array.emplaceBack.apply(array,
```

Comment:
I think this is a pretty hot path, so this change should receive some thorough benchmarking.